src/support/json.h

```diff
@@ -271,7 +280,13 @@ struct Value {
       assert(close);
       *close = 0; // end this string, and reuse it straight from the input
       char* raw = curr + 1;
-      unescapeAndSetString(raw);
+      if (stringEncoding == ASCII) {
+        // Just use the current string.
+        setString(raw);
```

Comment:
No unescaping at all? When would this non-compliant behavior be worth the presumably-tiny speedup? Decoding to WTF-8 instead would be ascii-compatible and fast when there are no escapes, but would still let us handle arbitrary escapes correctly for all the users currently using the ascii option.

Reply:
We use JSON for encoding the wasm-metadce data. That is simple ascii, without escaping. It's nice to avoid many small allocations there, though it's probably not a big speedup overall.